src/components/profile/ProfileDataTable.js

```diff
@@ -44,7 +44,7 @@ const ProfileDataTable = ({
   }, [verifyEdit, profile.email, storedProfile.email])
 
   const verifyPhone = useCallback(() => {
-    if (profile.mobile !== storedProfile.mobile) {
+    if (storedProfile.mobile && profile.mobile !== storedProfile.mobile) {
```

Comment:
the only issue with this is the case where a user can delete his phone number -> save -> then edit his profile and insert any number without verification

Reply:
i dont understand how does that fix anything, and why doesnt the case you described work.
i think we can call the verifymobile api endpoint here with the current mobile the user entered. if it returns already verified dont do anything, otherwise go to verifyedit and assume first SMS was sent.
Or make a new endpoint that just returns if mobile is verified or not